echopop/computation/biology.py

```diff
@@ -104,160 +128,751 @@ def sum_strata_weight( haul_data: pd.DataFrame ,
+
+    # ---- Calculate the overall sexed proportions taking into account aged and unaged fish (all age class)
+    stratum_sexed_kriged_biomass[ 'proportion_sexed_weight_all' ] = (
+        stratum_sexed_kriged_biomass.proportion_weight_sex *
+        stratum_sexed_kriged_biomass.proportion_unaged_weight_all
+    )
+
+    # ---- Calculate the overall sexed proportions taking into account aged and unaged fish (adults)
+    stratum_sexed_kriged_biomass[ 'proportion_sexed_weight_adult' ] = (
+        stratum_sexed_kriged_biomass.proportion_weight_sex *
+        stratum_sexed_kriged_biomass.proportion_unaged_weight_adult
+    )
+
+    # ---- Apportioned biomass for all fish
+    stratum_sexed_kriged_biomass[ 'biomass_sexed_unaged_all' ] = (
+        stratum_sexed_kriged_biomass.biomass_kriged
+        * stratum_sexed_kriged_biomass.proportion_sexed_weight_all
+        * stratum_sexed_kriged_biomass.proportion_weight_length
+    )
+
+    # ---- Apportioned biomass for adult fish
+    stratum_sexed_kriged_biomass[ 'biomass_sexed_unaged_adult' ] = (
+        stratum_sexed_kriged_biomass.biomass_kriged
+        * stratum_sexed_kriged_biomass.proportion_sexed_weight_adult
+        * stratum_sexed_kriged_biomass.proportion_weight_length
+    )
+
+    # ---- ' Grand total ' for all sexed unaged fish
+    apportioned_sexed_kriged_biomass = (
+            stratum_sexed_kriged_biomass
+            .groupby( [ 'species_id' , 'sex' , 'length_bin'  ] )
+            .agg( { 'biomass_sexed_unaged_all': 'sum' ,
+                    'biomass_sexed_unaged_adult': 'sum' } )
+            .reset_index( )
+        )
+    
+    ### Calculate biomass proportions across length
+    # ---- Initialize dataframe
+    stratum_kriged_biomass = indexed_proportions_unaged_weight_length
+    stratum_kriged_biomass[ 'proportion_unaged_weight_all' ] = indexed_unaged_proportions.proportion_unaged_weight_all
+    stratum_kriged_biomass[ 'proportion_unaged_weight_adult' ] = indexed_unaged_proportions.proportion_unaged_weight_adult
+    stratum_kriged_biomass[ 'biomass_kriged' ] = kriged_stratum_biomass
+
+    # ---- Apportioned biomass for all fish
+    stratum_kriged_biomass[ 'biomass_unaged_all' ] = (
+        stratum_kriged_biomass.biomass_kriged
+        * stratum_kriged_biomass. proportion_unaged_weight_all
+        * stratum_kriged_biomass.proportion_weight_length
+    )
+
+    # ---- Apportioned biomass for adult fish
+    stratum_kriged_biomass[ 'biomass_unaged_adult' ] = (
+        stratum_kriged_biomass.biomass_kriged
+        * stratum_kriged_biomass.proportion_unaged_weight_adult
+        * stratum_kriged_biomass.proportion_weight_length
+    )
+
+    # ---- ' Grand total ' for all sexed unaged fish
+    apportioned_kriged_biomass = (
+            stratum_kriged_biomass
+            .groupby( [ 'species_id' , 'length_bin' ] )
+            .agg( { 'biomass_unaged_all': 'sum' ,
+                    'biomass_unaged_adult': 'sum' } )
+            .reset_index( )
+        )
```

Comment:
Similar to my question in `compute_aged_biomass` [here](https://github.com/OSOceanAcoustics/echopop/pull/215/files?diff=unified&w=0#r1561905890):
wouldn't you be able to compute what's in `apportioned_kriged_biomass` by summing over all sexes in `apportioned_sexed_kriged_biomass`?